src/workerd/api/streams/standard.c++

```diff
@@ -1434,6 +1428,8 @@ void ReadableStreamDefaultController::enqueue(
     jsg::Optional<v8::Local<v8::Value>> chunk) {
   auto value = chunk.orDefault(js.v8Undefined());
 
+  JSG_REQUIRE(impl.canCloseOrEnqueue(), TypeError, "Unable to enqueue");
```

Comment:
Just a note, this wasn't quite what I expected when I read the commit message ("Do not call size algorithm in enqueue when stream is closed").